KeyError: 'outputs' when loading a big json plan output

I'm trying to test my module, but when I run `pytest` with my json I'm getting this error

```
fixtures_dir = '/home/guimaluf/workspace/rcs-as-code/test/fixtures'

    @pytest.fixture
    def plan_out(fixtures_dir):
        import json
        with open('%s/rcs_as_code.show.json' % fixtures_dir) as fp:
>           return tftest.TerraformPlanOutput(json.load(fp))

test_rcs_as_code.py:10:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tftest.TerraformPlanOutput object at 0x7f01785b12e0>
raw = {'configuration': {'provider_config': {'gitlab': {'expressions': {'base_url': {'constant_value': 'https://gitlab.devel...rces': [{'address': 'data.external.mcs94', 'mode': 'data', 'name': 'asd21', 'provider_name': 'external', ...}]}}}, ...}

    def __init__(self, raw):
      super(TerraformPlanOutput, self).__init__(raw)
      self.root_module = TerraformPlanModule(
          raw['planned_values']['root_module'])
>     self.outputs = TerraformValueDict(raw['planned_values']['outputs'])
E     KeyError: 'outputs'

lib/python3.8/site-packages/tftest.py:169: KeyError
```

I'm generating json with
```
terraform plan -out tfm.plan
terraform show -json tfm.plan > rcs_as_code.show.json
```

Using 
```Test session starts (platform: linux, Python 3.8.0, pytest 5.3.1, pytest-sugar 0.9.2)```

Please let me know If I need to provide additional info

--- a/.ci/cloudbuild.test.yaml
+++ b/.ci/cloudbuild.test.yaml
@@ -35,7 +35,7 @@ steps:
       - PATH=/usr/local/bin:/usr/bin:/bin:/builder/home/.local/bin
 
 substitutions:
-  _TERRAFORM_VERSION: 0.12.12
+  _TERRAFORM_VERSION: 0.12.19
 
 tags:
   - "ci"

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,7 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-### [Unreleased]
+## [Unreleased]
+
+### Fixed
+
+- fix `KeyError` on plan output when no Terraform outputs have been defined (cf [#9] and [#10])
 
 ## [1.3.0]
 

--- a/test/fixtures/no_outputs/main.tf
+++ b/test/fixtures/no_outputs/main.tf
@@ -1,0 +1,44 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+module "service-accounts" {
+  source     = "terraform-google-modules/service-accounts/google"
+  version    = "2.0.0"
+  project_id = var.project_id
+  prefix     = var.prefix
+  names      = var.names
+}
+
+module "gcs-buckets" {
+  source          = "terraform-google-modules/cloud-storage/google"
+  version         = "1.0.0"
+  project_id      = var.project_id
+  prefix          = var.prefix
+  names           = var.names
+  location        = var.gcs_location
+  set_admin_roles = true
+  bucket_admins = zipmap(
+    var.names,
+    module.service-accounts.iam_emails_list
+  )
+}
+
+resource "google_project_iam_member" "test_root_resource" {
+  project = var.project_id
+  role    = "roles/viewer"
+  member  = module.service-accounts.iam_email
+}

--- a/test/fixtures/no_outputs/variables.tf
+++ b/test/fixtures/no_outputs/variables.tf
@@ -1,0 +1,36 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+variable "gcs_location" {
+  description = "GCS bucket location."
+  type        = string
+}
+
+variable "names" {
+  description = "Resource names."
+  type        = list(string)
+}
+
+variable "prefix" {
+  description = "Prefix used for resource names."
+  type        = string
+}
+
+variable "project_id" {
+  description = "Id of the project used for resources."
+  type        = string
+}

--- a/test/test_no_outputs.py
+++ b/test/test_no_outputs.py
@@ -1,0 +1,45 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Test plan withno outputs."
+
+import pytest
+import tftest
+
+
+@pytest.fixture
+def plan(fixtures_dir):
+  tf = tftest.TerraformTest('no_outputs', fixtures_dir)
+  tf.setup(extra_files=['plan.auto.tfvars'])
+  return tf.plan(output=True)
+
+
+def test_variables(plan):
+  assert 'prefix' in plan.variables
+  assert plan.variables['names'] == ['one', 'two']
+
+
+def test_outputs(plan):
+  assert plan.outputs._raw == {}
+
+
+def test_root_resource(plan):
+  res = plan.resources['google_project_iam_member.test_root_resource']
+  assert res['values']['project'] == plan.variables['project_id']
+
+
+def test_modules(plan):
+  mod = plan.modules['module.gcs-buckets']
+  res = mod.resources['google_storage_bucket.buckets[0]']
+  assert res['values']['location'] == plan.variables['gcs_location']

--- a/tftest.py
+++ b/tftest.py
@@ -164,9 +164,10 @@ class TerraformPlanOutput(TerraformJSONBase):
 
   def __init__(self, raw):
     super(TerraformPlanOutput, self).__init__(raw)
+    planned_values = raw.get('planned_values', {})
     self.root_module = TerraformPlanModule(
-        raw['planned_values']['root_module'])
-    self.outputs = TerraformValueDict(raw['planned_values']['outputs'])
+        planned_values.get('root_module', {}))
+    self.outputs = TerraformValueDict(planned_values.get('outputs', {}))
     self.resource_changes = dict((v['address'], v)
                                  for v in self._raw['resource_changes'])
     self.variables = TerraformValueDict(raw['variables'])
@@ -188,7 +189,7 @@ class TerraformState(TerraformJSONBase):
 
   def __init__(self, raw):
     super(TerraformState, self).__init__(raw)
-    self.outputs = TerraformValueDict(raw['outputs'])
+    self.outputs = TerraformValueDict(raw.get('outputs', {}))
     self._resources = None
 
   @property

